### PR TITLE
Fix for fe-degree in VectorTools::project_boundary_values_curl_conforming_l2

### PR DESCRIPTION
**Pull request by winnifried**
This is the fix announced in https://groups.google.com/forum/#!topic/dealii/taFA8q2ugsg
for the application of VectorTools::project_boundary_values_curl_conforming_l2 to FE Systems 
containing multiple FE_Nedelec of different degree. 

It is tested in deal v 9.1.1 - I could not test in the current deal git version since dealii/dealii/master fails to compile in debug mode at the moment.

### CODE REVIEW

**Comment by masterleinad** [comment]
> It is tested in deal v 9.1.1 - I could not test in the current deal git version since dealii/dealii/master fails to compile in debug mode at the moment.

Would you mind opening another issue for this? Our regression tests (https://cdash.43-1.org/index.php?project=deal.II) don't indicate any problem, though.

**Comment by winnifried** [comment]
With the issue 
https://github.com/dealii/dealii/issues/8975
now resolved I could test the fix against dealii/dealii/master
and it fixes the first issue in https://groups.google.com/forum/#!topic/dealii/taFA8q2ugsg

**Comment by masterleinad** [comment]
Yes, that would be perfectly fine. Would you be willing to do that as well?
Just copy the file into `tests/numerics/project_bv_curl_conf_03.cc`, add the usual copyright header we have in every file, print "OK" at the end of the test, call `initlog()` at the start of the test and add  `tests/numerics/project_bv_curl_conf_03.output` containing the test output.
If that is too difficult, I could also do it.

**Comment by winnifried** [comment]
Sure, I will setup the test. May take a day due to other obligations.

File: source/numerics/vector_tools_integrate_difference.inst.in [hunk 1]
```diff
@@ -319,8 +319,7 @@ for (deal_II_dimension : DIMENSIONS; deal_II_space_dimension : SPACE_DIMENSIONS)
                            Vector<float>,
                            deal_II_space_dimension>(
         const DoFHandler<deal_II_dimension, deal_II_space_dimension> &,
-        const LinearAlgebra::TpetraWrappers::Vector<float>,
-          &,
+        const LinearAlgebra::TpetraWrappers::Vector<float> &,
```
Comment (bangerth) [inline source/numerics/vector_tools_integrate_difference.inst.in]: Bizarre that that would have compiled for anything at all before!

Comment (bangerth) [reply source/numerics/vector_tools_integrate_difference.inst.in]: Oh, I see -- it probably didn't but nobody uses Tretra to test...

**Comment by winnifried** [comment]
I have added a test for the fix. Let me know if you need anything else.

**Comment by bangerth** [comment]
Thanks, @winnifried and welcome to the authors of deal.II :-)

I should have mentioned this right away, but do you think you could add an entry to `doc/news/changes/` in the same style as the ones there? In a separate pull request, of course. This way, first we know that the bug was fixed, but we'll also list you as one of the people who contributed in the next release paper!

**Comment by winnifried** [comment]
Of course, the pull request is open.

**Pull request merged**